### PR TITLE
Consider using blobxfer for the drone azure plugin.

### PR DESCRIPTION
**Pull request by mnbbrown**
Hi all, 

How do you feel about using `blobxfer` for this plugin. It is a python script written by the Microsoft HPC and Azure batch teams. More details are available via [https://github.com/Azure/azure-batch-samples/tree/master/Python/Storage](https://github.com/Azure/azure-batch-samples/tree/master/Python/Storage) and
[http://blogs.technet.com/b/windowshpc/archive/2015/04/16/linux-blob-transfer-python-code-sample.aspx](http://blogs.technet.com/b/windowshpc/archive/2015/04/16/linux-blob-transfer-python-code-sample.aspx)

If so, I'd love to give implementing it a go.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -9,6 +14,57 @@ var (
...
+	}
+
+	cmd := command(vargs, workspace)
+	cmd.Env = os.Environ()
+	cmd.Dir = workspace.Path
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	trace(cmd)
+	err := cmd.Run()
```
Comment (bradrydzewski) [inline main.go:53]: tracing the command will expose the `StorageAccountKey` in the build logs. I would recommend tracing the command without `StorageAccountKey` and then adding `StorageAccountKey` before running the command.

or create two commands, once with `StorageAccountKey` and one without, and trace the command without.

**Comment by bradrydzewski** [comment]
one minor comment, but otherwise LGTM ... thanks!

**Comment by mnbbrown** [comment]
@bradrydzewski no worries - my first OSS contribution so I'm pretty happy! 

It now appends the credentials after the trace. I've also added a bit of logging in error handlers to help users debug builds.

**Comment by bradrydzewski** [comment]
LGTM

**Comment by mnbbrown** [comment]
Anything else this needs to be merged and made available as an plugin? 
It would be cool to be able to `beta` or `alpha` tag plugins.

File: DOCS.md [hunk 2]
```diff
@@ -1,13 +1,19 @@
-Use this plugin for publishing files and artifacts to Azure Storage. You can
-override the default configuration with the following parameters:
+Use this plugin for publishing files and artifacts to Azure Storage. The following parameters are required:
 
-* `` -
+* `account_key` - Storage Account Key, required for authentication
+* `storage_account` - Storage Account name
+* `container` - The target storage container
+* `source` - Tocation of folder to sync relative to the workspace root
```
Comment (tboerger) [inline DOCS.md]: Tocation?

**Comment by mnbbrown** [comment]
All fixed @tboerger :heavy_check_mark:

**Comment by mnbbrown** [comment]
The docker image is 200MB+ at the moment - I'm going to attempt to shrink it down to a more manageable size.

**Comment by tboerger** [comment]
The size is not a suprise with the packages you are installing. There are 3 options.
1. Build this plugin with pure go
2. Build and submit packages for the python libs to alpine
3. Uninstall the build deps after pip install

**Comment by tboerger** [comment]
If azure got a go ssk and you want to try it out i can build a go implementation today

**Comment by bradrydzewski** [comment]
I'm going to merge this. We can work on the size of the image in subsequent pull requests, but for now it is nice to have a working plugin

**Comment by bradrydzewski** [comment]
I'll post back when published to the registry

**Comment by mnbbrown** [comment]
Amazing - thanks @bradrydzewski.

Go implementation of blobxfer is the way to go I think - certainly will slim the whole thing down.
Azure do have a go SDK [https://github.com/Azure/azure-sdk-for-go](https://github.com/Azure/azure-sdk-for-go) but it's a mine field. I'm happy to write the go implementation - and submit another pull request.

**Comment by bradrydzewski** [comment]
ok, the plugin is now being automatically published to docker hub https://hub.docker.com/r/plugins/drone-azure-storage/

> I'm happy to write the go implementation - and submit another pull request.

awesome, for reference this is the google cloud storage implementation which uses channels for (faster) concurrent uploads https://github.com/drone-plugins/drone-google-cloudstorage

**Comment by mnbbrown** [comment]
Excellent - i'll have a look.

**Comment by tboerger** [comment]
:+1:

**Comment by mnbbrown** [comment]
I'll develop the plugin client here: https://github.com/mnbbrown/go-blobxfer
Aiming for it to be just a drop in replacement for the python implementation of blobxfer.

**Pull request merged**